activerecord/lib/active_record/autosave_association.rb

```diff
@@ -155,8 +155,25 @@ def self.valid_options
 
     module ClassMethods # :nodoc:
       private
+        if Module.method(:method_defined?).arity == 1 # MRI 2.5 and older
+          module MethodDefinedBackport
+            refine Module do
+              def method_defined?(method, inherit = true)
+                if inherit
+                  super(method)
+                else
+                  instance_methods(false).include?(name.to_sym)
+                end
+              end
+            end
+          end
+
+          using MethodDefinedBackport
```

Comment:
The house style seems to be to use an anonymous module for refinements:

https://github.com/rails/rails/blob/315f8b46038d0ce8ba5ec11e8909715b21be221b/activejob/lib/active_job/arguments.rb#L73
https://github.com/rails/rails/blob/315f8b46038d0ce8ba5ec11e8909715b21be221b/activesupport/lib/active_support/core_ext/enumerable.rb#L233
https://github.com/rails/rails/pull/40182

Reply:
Oh right, totally forgot.